conda_smithy/configure_feedstock.py

```diff
@@ -1328,6 +1330,23 @@ def render_README(jinja_env, forge_config, forge_dir, render_info=None):
             forge_config["azure"]["build_id"] = build_def["id"]
         except (IndexError, IOError):
             pass
+        except json.decoder.JSONDecodeError:
+            try:
+                # If it fails then we switch to a request using an Azure token.
+                from conda_smithy import azure_ci_utils
+
+                config = azure_ci_utils.AzureConfig(
+                    org_or_user=forge_config["azure"]["user_or_org"],
+                    project_name=forge_config["azure"]["project_name"],
+                )
+                repo = forge_config["github"]["repo_name"]
+                build_info = azure_ci_utils.get_build_id(repo, config)
+                forge_config["azure"]["build_id"] = build_info["build_id"]
+            except IOError:
+                # We don't want to command to fail if requesting the build_id fails.
+                logger.warning(
+                    "Azure build_id can't be retrieved using the Azure token."
+                )
```

Comment:
I would suggest just this small modification if you don't mind
```suggestion
            except IOError as err:
                # We don't want to command to fail if requesting the build_id fails.
                logger.warning(
                    "Azure build_id can't be retrieved using the Azure token. Exception: {}".format(err)
                )
```

Reply:
Make sense to me!